compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -552,12 +552,19 @@ impl MetaItemKind {
 
     fn from_mac_args(args: &MacArgs) -> Option<MetaItemKind> {
         match args {
+            MacArgs::Empty => Some(MetaItemKind::Word),
             MacArgs::Delimited(_, MacDelimiter::Parenthesis, tokens) => {
                 MetaItemKind::list_from_tokens(tokens.clone())
             }
             MacArgs::Delimited(..) => None,
-            MacArgs::Eq(_, token) => Lit::from_token(token).ok().map(MetaItemKind::NameValue),
-            MacArgs::Empty => Some(MetaItemKind::Word),
+            MacArgs::Eq(_, MacArgsEqKind::Ast(expr)) => {
+                if let ast::ExprKind::Lit(lit) = &expr.kind {
+                    Some(MetaItemKind::NameValue(lit.clone()))
+                } else {
+                    None
+                }
+            }
+            MacArgs::Eq(_, MacArgsEqKind::Hir(lit)) => Some(MetaItemKind::NameValue(lit.clone())),
```

Comment:
Is this case reachable?
I guess yes because meta-items can be created from decoded attributes.

Reply:
Yes, it occurs very soon during bootstrapping. Here's part of a stack trace:
```
  17:     0x7f9853e4e587 - <rustc_ast[eca82849642287e8]::ast::MetaItemKind>::from_mac_args
                               at /home/njn/dev/rust2/compiler/rustc_ast/src/attr/mod.rs:565:17
  18:     0x7f9853e4d2b9 - <rustc_ast[eca82849642287e8]::ast::AttrItem>::meta
                               at /home/njn/dev/rust2/compiler/rustc_ast/src/attr/mod.rs:227:19
  19:     0x7f9853e4d8ba - <rustc_ast[eca82849642287e8]::ast::Attribute>::meta
                               at /home/njn/dev/rust2/compiler/rustc_ast/src/attr/mod.rs:287:46
  20:     0x7f98521bffab - <rustc_passes[e862b3bd9538ab0d]::check_attr::CheckAttrVisitor>::check_doc_attrs
                               at /home/njn/dev/rust2/compiler/rustc_passes/src/check_attr.rs:926:27
  21:     0x7f98521bffab - <rustc_passes[e862b3bd9538ab0d]::check_attr::CheckAttrVisitor>::check_attributes
                               at /home/njn/dev/rust2/compiler/rustc_passes/src/check_attr.rs:87:29
  22:     0x7f98521c74e0 - <rustc_passes[e862b3bd9538ab0d]::check_attr::CheckAttrVisitor as rustc_hir[d52d3161296956a5]::intravisit::Visitor>::visit_item
                               at /home/njn/dev/rust2/compiler/rustc_passes/src/check_attr.rs:2044:9
  23:     0x7f9852239a66 - <rustc_hir[d52d3161296956a5]::intravisit::DeepVisitor<rustc_passes[e862b3bd9538ab0d]::check_attr::CheckAttrVisitor> as rustc_hir[d52d3161296956a5]::itemlikevisit::ItemLikeVisitor>::visit_item
```